allensdk/brain_observatory/filtered_running_speed/filtered_running_speed.py

```diff
@@ -0,0 +1,382 @@
+        # a 0 value for the change in orientation over an interval)
+        # there may be exact zeros in the velocity.
+        velocities = velocities[~(np.isclose(velocities["net_rotation"], 0.0))]
+
+        raw_data = pd.DataFrame(
+            {"vsig": vsig, "vin": vin, "frame_time": frame_times, "dx": dx}
+        )
+
+        return velocities, raw_data
+
+    def _process_single_simulus_experiment(
+        self
+    ):
+        """
+        Process an experiment with a single simulus sessions
+        """
+
+        start_index = FilteredRunningSpeed.START_FRAME
+
+        frame_times = self._get_frame_times()
+
+        end_index = len(frame_times)
+
+        velocities, raw_data = self._extract_dx_info(
+            frame_times,
+            start_index,
+            end_index,
+            self.pkl_path
+        )
+
+        store = pd.HDFStore(self.args['output_path'])
+        store.put("running_speed", velocities)
+        store.put("raw_data", raw_data)
+        store.close()
+
+        self._write_output_file()
+
+    def _write_output_file(self):
+        """
+        Write the output json file
+        """
+
+        ouput_data = {}
+        ouput_data['output_path'] = self.args['output_path']
+        ouput_data['input_parameters'] = self.args
+
+        with open(self.args['output_json'], 'w') as output_file:
+            json.dump(ouput_data, output_file, indent=2)
+
+    def _process_multi_simulus_experiment(
+        self
+    ):
+        """
+        Process an experiment with a three simulus sessions
+        """
+
+        (
+            behavior_start,
+            behavior_end,
+            mapping_start,
+            mapping_end,
+            replay_start,
+            replay_end
+        ) = self._get_stimulus_starts_and_ends()
+
+        frame_times = self._get_frame_times()
+
+        behavior_velocities = self._extract_dx_info(
+            frame_times,
+            behavior_start,
+            behavior_end,
+            self.args['behavior_pkl_path']
+        )
+
+        mapping_velocities = self._extract_dx_info(
+            frame_times,
+            mapping_start,
+            mapping_end,
+            self.args['mapping_pkl_path']
+        )
+
+        replay_velocities = self._extract_dx_info(
+            frame_times,
+            replay_start,
+            replay_end,
+            self.args['replay_pkl_path']
+        )
+
+        velocities, raw_data = self._merge_dx_data(
+            mapping_velocities,
+            behavior_velocities,
+            replay_velocities,
+            frame_times
+        )
+
+        store = pd.HDFStore(self.args['output_path'])
+        store.put("running_speed", velocities)
+        store.put("raw_data", raw_data)
+        store.close()
+
+        self._write_output_file()
+
+    def process(self, default_zscore_threshold, use_lowpass_filter):
+        """
+        Process an experiment
+        """
+
+        self.default_zscore_threshold = default_zscore_threshold
```

Comment:
Especially if these arguments get assigned to instance members, I'd prefer they be moved into the schema. I don't like the idea of methods altering the state of the instantiated class based on user input.

Reply:
Done.